scripts/set-react-version.js

```diff
@@ -98,35 +40,192 @@ function keys(obj) {
   return /** @type {(keyof T)[]} */ (Object.keys(obj));
 }
 
+/**
+ * Fetches the latest package information.
+ * @param {string} pkg
+ * @return {Promise<Manifest>}
+ */
+function fetchPackageInfo(pkg) {
+  return new Promise((resolve) => {
+    /** @type {string[]} */
+    const result = [];
+
+    const fetch = require("child_process").spawn("npm", [
+      "view",
+      "--json",
+      pkg,
+    ]);
+    fetch.stdout.on("data", (data) => {
+      result.push(data);
+    });
+    fetch.on("close", (code) => {
+      if (code !== 0 || result.length === 0) {
+        resolve({
+          version: undefined,
+          dependencies: {},
+          peerDependencies: {},
+        });
+      } else {
+        const list = JSON.parse(result.join(""));
+        const latest = Array.isArray(list) ? list[list.length - 1] : list;
+        resolve({
+          version: latest.version,
+          dependencies: latest.dependencies,
+          peerDependencies: latest.peerDependencies,
+        });
+      }
+    });
+  });
+}
+
+/**
+ * Returns a profile for specified version.
+ * @param {string} v
+ * @return {Promise<Record<string, string | undefined>>}
+ */
+async function getProfile(v) {
+  switch (v) {
+    case "canary-macos": {
+      const { dependencies, peerDependencies } = await fetchPackageInfo(
+        "react-native-macos@canary"
+      );
+      return {
+        "@react-native-community/cli":
+          dependencies["@react-native-community/cli"],
+        "@react-native-community/cli-platform-android":
+          dependencies["@react-native-community/cli-platform-android"],
+        "@react-native-community/cli-platform-ios":
+          dependencies["@react-native-community/cli-platform-ios"],
+        "hermes-engine": dependencies["hermes-engine"],
+        react: peerDependencies["react"],
```

Comment:
super nitpicky - but since this section is repeated equal for all the cases could it be refactored out?
```suggestion
        ...commonDepRefs,
```

Where
```
const commonDepRefs = {
   "@react-native-community/cli":
          dependencies["@react-native-community/cli"],
        "@react-native-community/cli-platform-android":
          dependencies["@react-native-community/cli-platform-android"],
        "@react-native-community/cli-platform-ios":
          dependencies["@react-native-community/cli-platform-ios"],
        "hermes-engine": dependencies["hermes-engine"],
        react: peerDependencies["react"],
}
```

or something? 😅